std/src/main/scala/algebra/std/boolean.scala

```diff
@@ -2,20 +2,45 @@ package algebra
 package std
 
 import algebra.lattice.Bool
+import algebra.ring.CommutativeRig
 
 package object boolean extends BooleanInstances
 
 trait BooleanInstances {
-  implicit val booleanAlgebra = new BooleanAlgebra
+  implicit val booleanAlgebra: BooleanAlgebra =
+    new BooleanAlgebra
 }
 
-class BooleanAlgebra extends Bool[Boolean] with Order[Boolean] {
+/**
+ * This commutative rig is different than the one obtained from GF(2).
+ *
+ * It uses || for plus, and && for times.
+ */
+class BooleanAlgebra extends Bool[Boolean] with Order[Boolean] with CommutativeRig[Boolean] {
   def compare(x: Boolean, y: Boolean): Int =
     if (x) { if (y) 0 else 1 } else { if (y) -1 else 0 }
```

Comment:
```scala
if (x == y) 0 else if (x) 1 else -1
```
might be my preferred way to go: for random data half the time you do only one comparison, worst case 2. Yours always does 2 (but with branch prediction who knows which would really be better).